raiden-ts/src/channels/reducer.ts

```diff
@@ -73,71 +73,39 @@ function channelOpenFailureReducer(
   return unset(path, state);
 }
 
-function channelDepositSuccessReducer(
+function channelUpdateOnchainBalanceStateReducer(
   state: RaidenState['channels'],
-  action: channelDeposit.success,
+  action: channelDeposit.success | channelWithdrawn,
 ): RaidenState['channels'] {
   const path = [action.meta.tokenNetwork, action.meta.partner];
   let channel: Channel | undefined = get(path, state);
   if (!channel || channel.state !== ChannelState.open || channel.id !== action.payload.id)
     return state;
-  if (action.payload.participant === action.meta.partner)
-    channel = {
-      ...channel,
-      partner: {
-        ...channel.partner,
-        deposit: action.payload.totalDeposit,
-      },
-    };
-  else
-    channel = {
-      ...channel,
-      own: {
-        ...channel.own,
-        deposit: action.payload.totalDeposit,
-      },
-    };
-  return set(path, channel, state);
-}
 
-function channelWithdrawReducer(
-  state: RaidenState['channels'],
-  action: channelWithdrawn,
-): RaidenState['channels'] {
-  const path = [action.meta.tokenNetwork, action.meta.partner];
-  let channel: Channel | undefined = get(path, state);
-  if (!channel || channel.state !== ChannelState.open || channel.id !== action.payload.id)
-    return state;
+  const key = channelWithdrawn.is(action) ? 'withdraw' : 'deposit';
+  const total = channelWithdrawn.is(action)
+    ? action.payload.totalWithdraw
+    : action.payload.totalDeposit;
+
   if (action.payload.participant === action.meta.partner)
     channel = {
       ...channel,
       partner: {
```

Comment:
Interesting approach on the `key`. Would it work for `partner`/`own` as well?

Reply:
Good point, I will adjust accordingly :)